frontend/catalyst/python_interface/inspection/xdsl_conversion.py

```diff
@@ -158,17 +172,33 @@ def _apply_adjoint_and_ctrls(qml_op: Operator, xdsl_op) -> Operator:
 
 
 # pylint: disable=too-many-return-statements
-def resolve_constant_params(ssa: SSAValue) -> float | int:
+def resolve_constant_params(ssa: SSAValue) -> float | int | str:
     """Resolve a constant parameter SSA value to a Python float or int."""
     op = ssa.owner
 
+    if isinstance(op, Block):
+        arg_name = list(compress(op.args, map(lambda arg: arg is ssa, op.args)))[0]
+        return arg_name.name_hint
+
     if isinstance(op, TensorExtractOp):
         return resolve_constant_params(op.tensor)
 
     if not hasattr(op, "name"):
         raise NotImplementedError(f"Cannot resolve parameters for operation: {op}")
 
     match op.name:
+        case "func.call":
+            if op.callee.string_value() == "remainder":
+                x = resolve_constant_params(op.operands[0])
+                y = resolve_constant_params(op.operands[1])
+                return f"({x} % {y})"
+            raise NotImplementedError(f"Function call to {op.callee} not supported")
+
+        case "tensor.from_elements":
+            return resolve_constant_params(op.operands[0])
+
+        case "arith.index_cast":
+            return resolve_constant_params(op.operands[0])
```

Comment:
I would suggest using this. 
```suggestion
        case "tensor.from_elements" | "arith.index_cast":
            return resolve_constant_params(op.operands[0])
```